### PR TITLE
[Merged by Bors] - TY-2657 handling sources

### PR DESCRIPTION
**Pull request by rustonaut**
**References:**

- [story TY-2528](https://xainag.atlassian.net/browse/TY-2528)
- [task TY-2657](https://xainag.atlassian.net/browse/TY-2657)

**Based on:**

- #275

### CODE REVIEW

File: discovery_engine/lib/src/ffi/types/init_config.dart [hunk 1]
```diff
@@ -62,14 +64,14 @@ class InitConfigFfi with EquatableMixin {
...
-    Set<String> excludedSources, {
+    Set<Source> excludedSources, {
     String? aiConfig,
   }) =>
       InitConfigFfi.fromParts(
         apiKey: configuration.apiKey,
         apiBaseUrl: configuration.apiBaseUrl,
         feedMarkets: configuration.feedMarkets.toList(),
-        excludedSources: excludedSources.toList(),
+        excludedSources: excludedSources.map((s) => s.toString()).toList(),
```
Comment (biernacki) [inline discovery_engine/lib/src/ffi/types/init_config.dart:74]: I've noticed it before, so maybe an extension would be a good idea, wdyt?
```dart
extension ToStringListExt on Set<Source> {
  List<String> toStringList() => map((s) => '$s').toList();
}

// then you could use it wherever you need to do that conversion 
excludedSources.toStringList();
```

**Review by biernacki — APPROVED** [review]
LGTM 🚀

**Comment by rustonaut** [comment]
bors merge

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**